service/service.go

```diff
@@ -108,6 +89,60 @@ func (svc *Service) Init(ctx context.Context, cfg *config.Config, buildTime, git
 	return nil
 }
 
+// initClients initializes external service clients based on the service configuration.
+// It sets up clients for Zebedee, DatasetAPI, and Topic services if their corresponding flags are enabled.
+func (svc *Service) initClients(ctx context.Context) {
+	if svc.Cfg.EnableZebedeeCallbacks {
```

Comment:
`svc.Cfg` could be nil (even though in the current context it never will be). We should probably check in order to avoid nil dereference errors. Also same for Dataset client.
```suggestion
	if svc.Cfg != nil && svc.Cfg.EnableZebedeeCallbacks {
```